radixdlt/src/main/java/org/radix/api/jsonrpc/SubmitAtomAndSubscribeEpic.java

```diff
@@ -95,32 +94,36 @@ public void onStored(boolean first) {
 			}
 
 			@Override
-			public void onError(AID atomId, Throwable e) {
-				if (e instanceof ParticleConflictException) {
-					ParticleConflictException particleConflictException = (ParticleConflictException) e;
-					ParticleConflict conflict = particleConflictException.getConflict();
-					JSONObject data = new JSONObject();
-					data.put("pointerToIssue", conflict.getDataPointer().toString());
-					data.put("message",
-							conflict.getAtomIds().stream().filter(a -> !a.equals(atomId)).findAny().map(Object::toString).orElse(null));
+			public void onStoredFailure(StoredFailure e) {
+				RadixEngineException exception = e.getException();
+				JSONObject data = new JSONObject();
+				data.put("pointerToIssue", exception.getDataPointer().toString());
```

Comment:
We ought to avoid using string literals for JSON keys... change into model object? :)

Reply:
no not worth it at the moment